src/plugins/esql_datagrid/public/data_grid.tsx

```diff
@@ -72,20 +73,22 @@ const DataGrid: React.FC<ESQLDataGridProps> = (props) => {
     setActiveColumns(columns);
   }, []);
 
-  const renderDocumentView = useCallback(
-    (
-      hit: DataTableRecord,
-      displayedRows: DataTableRecord[],
-      displayedColumns: string[],
-      customColumnsMeta?: DataTableColumnsMeta
-    ) => (
+  const renderDocumentView: UnifiedDataTableProps['renderDocumentView'] = useCallback(
+    ({
+      hit,
+      displayedRows,
```

Comment:
@jughosta can you help me understand the difference between displayedColumns and selected ones?

Reply:
@stratoula 

`columns` is what user selected in the sidebar and what we store in the app state (for example [] or ['agent'])
`displayedColumns` is what we actually render in the grid (for example ['@timestamp', '_source'] or ['@timestamp', 'agent'])

I had to pass `displayedColumns` to the flyout too so we can also show the time field when "Selected only" is switched on.

Reply:
Open for suggestions for `displayedColumns` prop. Maybe something like `columnsInGrid` would fit better.

Reply:
So the displayed columns are the table columns while the columns are the sidebar columns?

Reply:
@stratoula Yeah, here is an example where `columns` is `['extension', 'host', 'bytes']` and `displayedColumns` prop`['timestamp', 'extension', 'host', 'bytes']`

<img width="1888" alt="Screenshot 2024-09-05 at 13 20 19" src="https://github.com/user-attachments/assets/9757c3d7-a15f-449c-bf17-42b302ba313d">

Reply:
So the displayed columns are always the sidebar columns + timestamp if exists, right? If there is no sidebar columns it is [_source] + timestamp if exists. Is my logic correct?

Reply:
Yes @stratoula

Reply:
Can we have a property `displayTimeColumn: boolean` (or something like that) and construct the columns in the grid internally? In that case is a bit more obvious and we dont need 2 columns props which can be confusing.

If you think is not (you know the codebase better) then maybe renaming the prop to `columnsInDataGrid` and adding a description to each property can help overpass the confusion. Wdyt?

Reply:
I understand that it might be confusing and left many comments in the code explaining these 2 arrays.

`displayTimeColumn: boolean` could partially work for the feature but it does not help with knowing whether `_source` is shown or not and we need it for disabling the switch. 

@davismcphee wdyt?

Reply:
_source is shown when sidebarColumns.length === 0, no?

Reply:
Hmm yeah it's sort of weird either way and mainly just a symptom of the way we handle the time field and `_source` columns, which is confusing anyway. It might be slightly less confusing to use `displayTimeColumn` and check the columns length internally vs a second columns prop, but I don't have a strong opinion on it really. Ideally we'd clean up how we handle the special columns in general since similar issues exist elsewhere too, and maybe that's something we can once we allow toggling the `_source` column and deprecate the legacy grid.

Reply:
Once we allow toggling for `_source` we would also need to add another similar properly like `displaySource` to disable the switch. I still see a benefit of having access to the final list of columns instead of supporting separate props per special column or at least combine this information into a single prop, for example:

specialColumnsInfo={{
  timeFieldColumn: '@timestamp',
  hasSourceColumn: true
}}

So it would be easier to extend in the future. Wdyt?

Reply:
I'm a bit confused by the situation I think. My understanding was that we currently pass through the user selected columns (so everything except time field and `_source`). Would something like this not work?
- Always filter out `_source` and `dataView.timeField` from the passed columns.
- If filtered columns array is empty, disable selected fields toggle.
- If filtered columns array is not empty, enable selected fields toggle.
- If selected fields toggle is on, always include `dataView.timeField` in the results.

Maybe I'm missing something here though? And in either case, I still stand by my approval 😄 I'm ok with merging this with both columns props if there are no strong feelings otherwise. We're limited in how clean we can make this due to the funny handling of special columns regardless.

Reply:
I have the same feeling with Davis, I really find the 2 columns properties super confusing, the comments help but still someone needs to remember how Discover works to understand what is going on. If I was the code owner I would suggest to go with one columns property but I am not so I leave this decision to you.

Reply:
I refactored the approach by removing `displayedColumns` array and extracting utils around time field handling from `unified-data-table` to `discover-utils` package so I could reuse them in `unified-doc-viewer` https://github.com/elastic/kibana/pull/191930/commits/89832906dbd71dd7b4d122600d4f5c46df2c9519